eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use Exception;
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\MVC\Symfony\Cache\GatewayCachePurger;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+/**
+ * The ezplatform:urls:regenerate-aliases Symfony command implementation.
+ * Recreates system URL aliases for all existing Locations and cleanups corrupted URL alias nodes.
+ */
+class RegenerateUrlAliasesCommand extends Command
+{
+    const DEFAULT_ITERATION_COUNT = 1000;
+
+    /**
+     * @var \eZ\Publish\API\Repository\Repository
+     */
+    private $repository;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\Cache\GatewayCachePurger
+     */
+    private $cachePurger;
+
+    /**
+     * @param \eZ\Publish\API\Repository\Repository $repository
+     * @param \eZ\Publish\Core\MVC\Symfony\Cache\GatewayCachePurger $cachePurger
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        Repository $repository,
+        GatewayCachePurger $cachePurger,
+        LoggerInterface $logger = null
+    ) {
+        parent::__construct();
+
+        $this->repository = $repository;
+        $this->cachePurger = $cachePurger;
+        $this->logger = null !== $logger ? $logger : new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:urls:regenerate-aliases')
+            ->setDescription(
+                'Regenerates Location URL aliases (autogenerated) and cleans up custom Location ' .
+                'and global URL aliases stored in the Legacy Storage Engine'
+            )
+            ->addOption(
+                'iteration-count',
+                'c',
+                InputOption::VALUE_OPTIONAL,
+                'Number of Locations fetched into memory and processed at once',
+                self::DEFAULT_ITERATION_COUNT
+            )
+            ->setHelp(
+                <<<EOT
+<error>Make sure to back up your database before proceeding.</error>
```

Comment:
nitpick: warning? But this can work as well, just a but strange to see usage of error on something which is not a error condition :P

Reply:
Maybe also group the two errors/warnings togheter at the top.

For instance:
```
Before you continue:
- Make sure to back up your database before proceeding.
- Take installation offline, during the script execution the database should not be modified.
```

Reply:
> nitpick: warning? But this can work as well, just a but strange to see usage of error on something which is not a error condition :P

Its about the color used for color-capable terminals. There's no `warning` formatting tag. See [Using color styles](https://symfony.com/doc/2.8/console/coloring.html#using-color-styles) Sf doc. We could use some other tag, but red seems like the best choice chere.

> Maybe also group the misc errors/warnings together at the top.

Good idea. I expanded on it and reused the same information during command runtime, before user decides to continue processing Locations. Fixed via 01bb253.